chain-impl-mockchain/src/vote/manager.rs

```diff
@@ -434,24 +455,34 @@ impl ProposalManagers {
 
     /// validate the vote against the proposal: verify that the proposal exists
     /// and the the length of the ciphertext is correct (if applicable)
-    pub fn validate_vote(&self, cast: &VoteCast) -> Result<(), VoteError> {
+    pub fn validate_vote(
+        &self,
+        cast: VoteCast,
+        vote_plan: &VotePlan,
+    ) -> Result<(ValidatedVote, usize), VoteError> {
```

Comment:
Does the proposal index need to be returned? The caller can get it from `cast` if necessary.

Reply:
It was a way to encourage the user to use the validated proposal index and not something external, but agree that it's already available